online_meetings/report.py

```diff
@@ -36,27 +37,50 @@
 DEFAULT_SLEEP_TIME = ENV.get('DEFAULT_SLEEP_TIME', 10)
 
 
-# Functions
-def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = None, params: Dict[str, Union[str, int]] = None):
-    if params is None:
-        params = {}
-    if headers is None:
-        headers = {}
+def get_request_retry(url: str, headers: Dict[str, Union[str, int]],
+                      params: Dict[str, Union[str, int]]) -> requests.Response:
+
+    response = requests.request("GET", url, headers=headers, params=params)
+
+    # Rate limited, wait a few seconds
+    if response.status_code == requests.codes.too_many_requests:
+        # This is what the header should be
+        retry_after = response.headers.get("Retry-After")
+        sleep_time = DEFAULT_SLEEP_TIME
+        if retry_after and retry_after.isdigit():
+            logger.warning(
+                f"Received status 429, need to wait for {retry_after}")
+            sleep_time = retry_after
+        else:
+            logger.warning(
+                f"No Retry-After header, setting sleep loop for {DEFAULT_SLEEP_TIME} seconds")
+        while response.status_code == requests.codes.too_many_requests:
+            time.sleep(sleep_time)
+            response = requests.request(
+                "GET", url, headers=headers, params=params)
+
+    # If it's not okay at this point, raise an error
+    if response.status_code != requests.codes.ok:
+        logger.warning(f'Received irregular status code: {response.status_code}')
+        response.raise_for_status()
```

Comment:
May not need this if you're going to do the same thing just outside of this function.

Reply:
Wait, yes, you do. Maybe the logging warning could go away, since you seem to be logging in the `try/except` clauses that wrap uses of this function.

Reply:
Yeah sounds good, I had it in there before I refactored it to throw an exception. This seemed to be the preferred pattern for handling these response errors.